### PR TITLE
Add "goal bot" diagrams

### PR DESCRIPTION
**Pull request by DannyNicholas**
# Description

To avoid manually editing architecture diagrams, I've created these using PlantUML so we have "diagrams-as-code". The diagrams represent the early version of "goal-bot" being used to gradually expand InferGPT features.

This PR holds the PlantUML code and the generated diagrams. 

I did investigate using Mermaid but producing nice-looking charts was a lot more work. PlantUML charts look good without any additional styling and I found them simpler to maintain.

See: https://plantuml.com/

Review `README` at: https://github.com/WaitThatShouldntWork/InferGPT/tree/feature/goal-bot-diagrams/assets/goal-bot

---

# Changelog
- Add PlantUML code and generated images for: architecture and chain-of-command
- Add `README` to reference the diagrams in this folder.

### CODE REVIEW

File: assets/goal-bot/GoalBotArchitecture.iuml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+'
+' PlantUML diagram for the Goal Bot architecture.
+' Please re-generate the image (.png) after updating the code
```
Comment (harryBedfordSL) [inline assets/goal-bot/GoalBotArchitecture.iuml:3]: Forgive me if I've missed something, but how do we generate the image from the iuml?

Comment (DannyNicholas) [reply assets/goal-bot/GoalBotArchitecture.iuml:3]: @harryBedfordSL - Good question. I'll document that in the README. I'm using Visual Code to do this for me.

Comment (Ch0live) [reply assets/goal-bot/GoalBotArchitecture.iuml:3]: Looks like you can do it with a command

java -jar plantuml.jar ".../assets/goal-bot"

but [requires Java 8 or onwards](https://plantuml.com/command-line)

File: assets/goal-bot/ChainOfCommand.iuml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'
+' PlantUML diagram for Chain of Command.
+' Please re-generate the image (.png) after updating the code
+'
+@startuml ChainOfCommand
+
+(Director) <-> (Supervisors)
+(Supervisors) <-> (Agent)
+(Agent) <-> (Tools)
```
Comment (Ch0live) [inline assets/goal-bot/ChainOfCommand.iuml:9]:
```suggestion
(Supervisors) <-> (Agents)
(Agents) <-> (Tools)
```

File: assets/goal-bot/GoalBotArchitecture.iuml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+director --> supervisor
+director ..> llm
+Agents ..> llm
+supervisor --> goals
+supervisor --> nextBestQuestion
+supervisor --> summariser
+goals --> graphUtils
+graphUtils --> data
+
+@enduml
```
Comment (Ch0live) [inline assets/goal-bot/GoalBotArchitecture.iuml:65]: I love this sorta stuff, great job 🚀

File: assets/goal-bot/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+This folder holds documentation and design related to "goal-bot".
+
+This is a very early version of InferGPT that is being used to gradually expand the features supported. When reviewing the intended long-term design, it is recommended to review the top-level [README](../README.md).
+
+## Architecture
+
+High Level Architecture of the initial version of "goal-bot"
+
+> This is work-in-progress and will actively change.
```
Comment (Ch0live) [inline assets/goal-bot/README.md:13]: Cool GitHub markdown thing if interested, you can make a Note section here that pops out on the rendered few like this 

```suggestion
> [!NOTE]
> This is work-in-progress and will actively change.
```

See more info [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)

File: assets/goal-bot/GoalBotArchitecture.iuml (hunk 1 above)
Comment (DannyNicholas) [reply assets/goal-bot/GoalBotArchitecture.iuml:3]: @Ch0live  / @harryBedfordSL - thanks for feedback. I've recommending using the IDE extension. Being a python project, not every developer with have Java or the plantuml JAR file.

**Review by Ch0live — APPROVED** [review]

**Pull request merged**